beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/filter/WhitelistedListenerEventFilter.java

```diff
@@ -32,24 +31,26 @@
 
   @Override
   public boolean filter(ListenerEvent listenerEvent) {
-    Class<? extends ListenerEvent> eventClass = listenerEvent.getEventType().eventClass();
-    if (AlterPartitionEvent.class.equals(eventClass) || AlterTableEvent.class.equals(eventClass)) {
-      return false;
-    }
     Map<String, String> tableParameters = listenerEvent.getTableParameters();
-    if (tableParameters == null) {
-      return true;
+    if (tableParameters != null && tableParameters.get(BEEKEEPER_HIVE_EVENT_WHITELIST) != null) {
+      return !isWhitelisted(listenerEvent, tableParameters.get(BEEKEEPER_HIVE_EVENT_WHITELIST));
     }
-    return !isWhitelisted(listenerEvent.getEventType(), tableParameters);
+    return !isDefaultBehaviour(listenerEvent);
   }
 
-  private boolean isWhitelisted(EventType eventType, Map<String, String> tableParameters) {
-    String whitelist = tableParameters.get(BEEKEEPER_HIVE_EVENT_WHITELIST);
-    if (whitelist == null) {
-      return false;
-    }
+  private boolean isWhitelisted(ListenerEvent listenerEvent, String whitelist) {
     return Arrays.stream(whitelist.split(","))
       .map(String::trim)
-      .anyMatch(whitelistedEvent -> whitelistedEvent.equalsIgnoreCase(eventType.toString()));
+      .anyMatch(whitelistedEvent -> whitelistedEvent.equalsIgnoreCase(listenerEvent.getEventType()
+        .toString()));
+  }
+
+  private boolean isDefaultBehaviour(ListenerEvent listenerEvent) {
+    Class<? extends ListenerEvent> eventClass = listenerEvent.getEventType()
```

Comment:
Quick question: is class comparison the best/only way to perform this check? Can it be done with `instanceof` instead or do you need exact class comparison?